### ISSUE TITLE
Random crash

### ISSUE THREAD

**Issue by libcg** [libcg/gSquare#1]
There's two threads, one for game logic and rendering. They are currently unsynchronized, and unlike the PSP, modern operating systems use preemption and that makes the game crash. Fix this issue with mutexes.

**Comment by libcg** [comment]
Should be fixed by 7239415. Closing.

**Comment by libcg** [comment]
Still crashing for some reason.

**Comment by libcg** [comment]
Huge issue with font rendering, GL textures are created on the fly but never free'd:
```
g2dBeginRects(g2dTexFromFont(bigfont,timer_text,WHITE));
...
g2dEnd();
```
Depends on issue #3.

**Comment by libcg** [comment]
Hopefully fully fixed by acd92196.
